main.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  # each
+  def my_each
+    each do |i|
+      yield i
+    end
+  end
+
+  # each_with_index
+  def my_each_with_index
+    result = map do |element, index|
+      yield "#{index}: #{element} "
+    end
+    result
+  end
+
+  # select
+  def my_select
+    result = []
+    each do |i|
+      result << i if yield(i)
+    end
+    result
+  end
+
+  # all
+  def my_all?
+    my_each do |_elememt|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  # any
+  def my_any?
+    my_each do |_elememt|
+      return true if yield(element)
+    end
+    false
+  end
```

Comment:
- [ ] `my_any?` returns true if the block ever returns a value other than false or nil
`p [12,12,13,1,4].my_any? { |num| num <= 3 }` should return true

- [ ] `my_any?` when no block or argument is given returns true if at least one of the collection is not false or nil
`p [nil,true, nil].my_any?` should return true

- [ ] `my_any?` when a class is passed as an argument returns true if at least one of the collection is a member of such class
`p [12,3,34].my_any?(Integer)` should return true

- [ ] `my_any?` when a Regex is passed as an argument returns false if none of the collection matches the Regex
`p ['nice','awesome'].my_any?(/z/)` should return false

- [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
`p ['dog',6,99,].my_any?('car')` should return false